src/Database/LSMTree/Common.hs

```diff
@@ -172,7 +242,13 @@ openSession hfs hbio dir = do
 -- lock will be released).
 --
 closeSession :: IOLike m => Session m -> m ()
-closeSession = undefined
+closeSession (Session (Internal.Session{Internal.sessionState})) =
+    RW.modify_ sessionState $ \case
+      Internal.SessionClosed -> pure Internal.SessionClosed
+      Internal.SessionOpen (Internal.SessionEnv {Internal.sessionFileLock}) -> do
+        -- TODO: close known table handles
```

Comment:
I'm now reconsidering whether we should guarantee that table handles are closed when their session is closed. Maybe it's just simpler to tell the user: close all your handles before you close your session, otherwise you get exceptions